meson.build

```diff
@@ -49,6 +49,9 @@ endif
 
 conf.set('HAVE_LOCALE_T', enable_locale and cc.has_header_symbol('locale.h', 'locale_t'))
 conf.set('HAVE_ICONV', enable_locale and cc.has_function('iconv'))
+if cc.has_function('iconv')
+  iconv_dep = dependency('iconv')
```

Comment:
If cc.has_function('iconv') then it is built into the libc and you do NOT need an external dependency for it.

`dependency('iconv')` is meaningless in the general sense, because there's no pkg-config file for it. It is documented in https://mesonbuild.com/Dependencies.html#iconv to be available starting Meson 0.60.0 and handles checking for cc.has_function internally.

Reply:
Without declaring iconv dependency, I got linking error on Termux

```
ld.lld: error: undefined symbol: libiconv_open
>>> referenced by charset.cxx:108 (../src/charset.cxx:108)
>>>               ncmpc.p/src_charset.cxx.o:(CopyUtf8ToLocale(char*, unsigned long, char const*, unsigned long))
>>> referenced by charset.cxx:171 (../src/charset.cxx:171)
>>>               ncmpc.p/src_charset.cxx.o:(Iconv(char const*, char const*, char const*, unsigned long))

ld.lld: error: undefined symbol: libiconv
>>> referenced by charset.cxx:66 (../src/charset.cxx:66)
>>>               ncmpc.p/src_charset.cxx.o:(CopyUtf8ToLocale(char*, unsigned long, char const*, unsigned long))
>>> referenced by charset.cxx:132 (../src/charset.cxx:132)
>>>               ncmpc.p/src_charset.cxx.o:(Iconv(char const*, char const*, char const*, unsigned long))

ld.lld: error: undefined symbol: libiconv_close
>>> referenced by charset.cxx:114 (../src/charset.cxx:114)
>>>               ncmpc.p/src_charset.cxx.o:(CopyUtf8ToLocale(char*, unsigned long, char const*, unsigned long))
>>> referenced by charset.cxx:175 (../src/charset.cxx:175)
>>>               ncmpc.p/src_charset.cxx.o:(Iconv(char const*, char const*, char const*, unsigned long))
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```
For now, I added `required: false` for finding iconv library, so configuration won't fail when iconv is part of libc instead of separate shared library.